### PR TITLE
Add filters list to the resource view of a Gateway

### PR DESCRIPTION
**Pull request by AndrewDryga**
Closes https://github.com/firezone/product/issues/654

### CODE REVIEW

File: elixir/apps/api/test/api/gateway/channel_test.exs [hunk 1]
```diff
@@ -89,7 +89,11 @@ defmodule API.Gateway.ChannelTest do
                name: resource.name,
                type: :dns,
                ipv4: resource.ipv4,
-               ipv6: resource.ipv6
+               ipv6: resource.ipv6,
+               filters: [
+                 %{ports: ["80", "433"], protocol: :tcp},
+                 %{ports: ["100 - 200"], protocol: :udp}
```
Comment (AndrewDryga) [inline elixir/apps/api/test/api/gateway/channel_test.exs]: @conectado if this format is hard to parse we can render it differently. Currently, it's either a number as a string `"80"` or range `"80 - 8080"` (and it's how we also store it in DB).

Comment (conectado) [reply elixir/apps/api/test/api/gateway/channel_test.exs]: Could we do an structured format, like `max`, `min` and `porotocol`?

Comment (AndrewDryga) [reply elixir/apps/api/test/api/gateway/channel_test.exs]: @conectado min/max one doesn't cover case where you want to explicitly set multiple ports ("80, 433"), but I can unroll those for you turning it into `%{min: 80, max: 80, protocol: :tcp}, %{min: 433, max: 433, protocol: :tcp}`. Would that be preferred?

Comment (conectado) [reply elixir/apps/api/test/api/gateway/channel_test.exs]: Yeah I think it's better

Comment (AndrewDryga) [reply elixir/apps/api/test/api/gateway/channel_test.exs]: @conectado done:
```
%{protocol: :tcp, port_range_end: 80, port_range_start: 80},
%{protocol: :tcp, port_range_end: 433, port_range_start: 433},
%{protocol: :udp, port_range_start: 100, port_range_end: 200}
```

**Review by conectado — APPROVED** [review]
